examples/async/hello_world.ml

```diff
@@ -22,9 +21,10 @@ let handler ~body:_ _sock req =
 let start_server port () =
   Caml.Printf.eprintf "Listening for HTTP on port %d\n" port;
   Caml.Printf.eprintf "Try 'curl http://localhost:%d/test?hello=xyz'\n%!" port;
-  Cohttp_async.Server.create ~on_handler_error:`Raise
-    (Async.Tcp.Where_to_listen.of_port port) handler
-  >>= fun _ -> Deferred.never ()
+  let _never, server = Cohttp_async.Server.create ~on_handler_error:`Raise
+    ~protocol:Conduit_async.TCP.protocol ~service:Conduit_async.TCP.service
+    (Conduit_async.TCP.Listen (Async.Tcp.Where_to_listen.of_port port)) handler in
+  server
```

Comment:
How is this thing of `_never, server`, returning `server, instead of throwing away everything with `fun _ -> Deferred.never ()` working?

Reply:
We can imagine to replace my update by `let _, server = ... in server >>= fun _ -> Deferred.never ()`. `Cohttp_async.Server.create` does not returns an `Async` value but a pair of an `Async` condition and an `Async` value. The user must destruct it and use `server` to _launch_ the server (however, in reality, the server is already launched). `_never` is the condition to stop the server and in this special case, we don't want to stop the server - so I added `_` in front of `never`.

Reply:
Cool, thanks!